### PR TITLE
Add unit tests for the benchmark repo

### PR DESCRIPTION
**Pull request by berroar**
### What is the context of this PR?

Adds unit tests for our scripts in our benchmark repo, including a mock output directory. The benchmark repo was initially created without any tests, so it is difficult to gain 100% coverage as a first pass. 

We could look at amending some of the logic in future, for example, extracting some of the code from __main__, which I have done as part of this PR in `visualise_results.py` to enable better testing. There are some methods that still have no coverage, for example the google_cloud_storage get_files method, as although it is possible to mock the GCS client is difficult to add meaningful tests.

Adds the tests to our actions workflow on PR and enforces coverage at 80%.

Adds isort to benchmark and runs as part of `make format`.

### How to review

Run the unit tests with `make test`. Check the tests added to make sure they make sense, can any other tests be added?

Check the benchmark is still working.

### CODE REVIEW

**Comment by berroar** [comment]
[Add tests for eq-survey-runner-benchmark scripts (3 SP)](https://trello.com/c/WQxCvEQ0/5948-add-tests-for-eq-survey-runner-benchmark-scripts-3-sp)

File: scripts/visualise_results.py [hunk 1]
```diff
@@ -8,35 +8,36 @@
...
+    try:
+        plt.style.use('fast')
 
-    plt.savefig('performance_graph.png', bbox_inches="tight")
-    print("Graph saved as performance_graph.png")
+        if (
+            number_of_days_to_plot and number_of_days_to_plot <= 45
+        ):  # To make the chart still easily digestible
+            df.plot.line(marker="o", markersize=8)
+            # plt.grid(True, axis="both", alpha=0.3)
```
Comment (rmccar) [inline scripts/visualise_results.py:23]: Are we keeping this line even though its commented out?

**Review by rmccar — COMMENTED** [review]
Other than this looks good

File: scripts/visualise_results.py (hunk 1 above)
Comment (rmccar) [reply scripts/visualise_results.py]: Yeah thought it must've been something like that

**Review by rmccar — APPROVED** [review]

**Review by VirajP1002 — APPROVED** [review]

File: Pipfile [hunk 2]
```diff
@@ -17,6 +18,10 @@ google-cloud-storage = "*"
 pandas = "*"
 matplotlib = "*"
 slackclient = "*"
+pytest = "*"
```
Comment (liamtoozer) [inline Pipfile:21]: Does the pytest dependencies need to go into the dev-packages?

File: tests/test_visualise_results.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    mock_plot_data(dataframe, 1)
+
+    assert mock_plot_data.call_count == 1
+    assert_frame_equal(mock_plot_data.call_args[0][0], expected_data_frame)
+
+
+def test_plot_data(get_results_single_file):
+    dataframe = get_data_frame(get_results_single_file)
+    try:
+        plot_data(dataframe, 1)
```
Comment (liamtoozer) [inline tests/test_visualise_results.py:52]: Quick couple of questions - Does this test need an assertion? Also, is it worth removing the file that gets generated here, or mocking the call and asserting that it's called?

File: scripts/visualise_results.py [hunk 4]
```diff
@@ -8,35 +8,36 @@
...
 
-    plt.savefig('performance_graph.png', bbox_inches="tight")
-    print("Graph saved as performance_graph.png")
+        if (
+            number_of_days_to_plot and number_of_days_to_plot <= 45
+        ):  # To make the chart still easily digestible
+            df.plot.line(marker="o", markersize=8)
+            plt.grid(True, axis="both", alpha=0.3)
+        else:
+            df.plot.line()
```
Comment (liamtoozer) [inline scripts/visualise_results.py:25]: Another quick question! I noticed that this condition is missing test coverage, is this one of the bits that can't be tested?

File: Pipfile (hunk 2 above)
Comment (berroar) [reply Pipfile]: Moved these now 👍

File: tests/test_visualise_results.py (hunk 3 above)
Comment (berroar) [reply tests/test_visualise_results.py:53]: I had a look and it seems like one of the only ways to test the graph generation is by using a library that is able to compare the graph generated in the test against a stored image. But spoke with Mebin and we agreed it was enough to test that the plot_data method does not fail when it is passed a valid dataframe for now. However as you pointed it was generating the graph when running the test, so I have mocked that and added an assertion that the `savefig` method is called 👍

File: scripts/visualise_results.py (hunk 4 above)
Comment (berroar) [reply scripts/visualise_results.py:25]: This is one that needs to be looked at separately, i think we will raise a card to look at refactoring some of the benchmark scripts now that we have test coverage. Not sure that we ever use this else anymore.

File: tests/test_visualise_results.py (hunk 3 above)
Comment (liamtoozer) [reply tests/test_visualise_results.py:53]: Ah cool that makes sense, thanks!

File: scripts/visualise_results.py (hunk 4 above)
Comment (liamtoozer) [reply scripts/visualise_results.py:25]: That makes sense, thanks 👍

**Review by liamtoozer — APPROVED** [review]
Tested locally and works great for me 👍

File: Pipfile [hunk 5]
```diff
@@ -17,6 +21,8 @@ google-cloud-storage = "*"
 pandas = "*"
 matplotlib = "*"
 slackclient = "*"
+freezegun = "*"
+numpy = "*"
```
Comment (MebinAbraham) [inline Pipfile:25]: These should be dev packages.

Comment (berroar) [reply Pipfile]: Moved freezegun, removed numpy as I think I added it in error in the first place, it is already installed with pandas and not needed for the tests 👍

**Pull request merged**